OutOfSchool/OutOfSchool.WebApi.Tests/Validators/CustomUkrainianNameAttributeTests.cs

```diff
@@ -0,0 +1,139 @@
+    ];
+
+    public static IEnumerable<string> NonUkrainianSingleNameStrings =>
+    [
+        "Валер'ian", "Veler'ян",
+        "Anastasia", "Mariia", "Vladyslav", "Dmytro",
+        "明美", "明里", "愛子", "金锦津錦", "金澤", "김연아",
+        "Рыбаков", "Рыбак", "Объектов", "Объект", "Ёженов", "Проёжов", "Эвклид", "Проэкт",
+    ];
+
+    public static IEnumerable<object[]> ValidComminSingleNamesTestParams =>
+        ValidCommonSingleNamesStrings.Select(n => new object[] { n });
+
+    public static IEnumerable<object[]> ValidCommonDoubleNamesTestParams =>
+        ValidCommonSingleNamesStrings.Select(n => new object[] { $"{n}-{n}" });
+
+    public static IEnumerable<object[]> InvalidNonUkrainianSingleNamesTestParams =>
+        NonUkrainianSingleNameStrings.Select(n => new object[] { n });
+
+    [Test]
+    public void IsValid_WhenNameIsNull_ShouldReturnTrue()
+    {
+        // Arrange
+        var value = null as string;
+
+        // Act
+        var isValid = new CustomUkrainianNameAttribute().IsValid(value);
+
+        // Assert
+        Assert.IsTrue(isValid);
+    }
+
+    [Test]
+    public void IsValid_WhenNameIsEmpty_ShouldReturnTrue()
+    {
+        // Arrange
+        var value = string.Empty;
+
+        // Act
+        var isValid = new CustomUkrainianNameAttribute().IsValid(value);
+
+        // Assert
+        Assert.IsTrue(isValid);
+    }
+
+    [Test]
+    public void IsValid_WhenNameIsWhiteSpace_ShouldReturnTrue()
+    {
+        // Arrange
+        var value = " ";
+
+        // Act
+        var isValid = new CustomUkrainianNameAttribute().IsValid(value);
+
+        // Assert
+        Assert.IsTrue(isValid);
+    }
+
+    [TestCaseSource(nameof(ValidComminSingleNamesTestParams))]
+    public void IsValid_WhenNameIsValidCommonSingleName_ShouldReturnTrue(object value)
+    {
+        // Act
+        var isValid = new CustomUkrainianNameAttribute().IsValid(value);
+
+        // Assert
+        Assert.IsTrue(isValid);
+    }
+
+    [TestCaseSource(nameof(ValidCommonDoubleNamesTestParams))]
+    public void IsValid_WhenNameIsValidCommonDoubleName_ShouldReturnTrue(object value)
+    {
+        // Act
+        var isValid = new CustomUkrainianNameAttribute().IsValid(value);
+
+        // Assert
+        Assert.IsTrue(isValid);
+    }
+
+    [Test]
+    public void IsValid_WhenNameIsNotString_ShouldReturnFalse()
+    {
+        // Arrange
+        var value = DateTime.UtcNow;
+
+        // Act
+        var isValid = new CustomUkrainianNameAttribute().IsValid(value);
+
+        // Assert
+        Assert.IsFalse(isValid);
+    }
+
+    [TestCaseSource(nameof(InvalidNonUkrainianSingleNamesTestParams))]
+    public void IsValid_WhenNameIsNonUkrainianName_ShouldReturnFalse(object value)
+    {
+        // Act
+        var isValid = new CustomUkrainianNameAttribute().IsValid(value);
+
+        // Assert
+        Assert.IsFalse(isValid);
+    }
+
+    [Test]
+    public void IsValid_WhenNameIsContainsDigits_ShouldReturnTrue()
```

Comment:
(Mistype) Test name `IsValid_WhenNameIsContainsDigits_ShouldReturnTrue` has `...IsContains...`. Please make it more perfect.